test/cppsuite/test_harness/workload_tracking.h

```diff
@@ -0,0 +1,79 @@
+#ifndef WORKLOAD_TRACKING_H
+#define WORKLOAD_TRACKING_H
+
+#include <iostream>
+
+/* Default schema for tracking table
+ * key_format : Collection name / Key
+ * value_format : Operation type / Value / Time
+ */
+#define DEFAULT_TRACKING_TABLE_SCHEMA "key_format=Si,value_format=iSi"
+
+namespace test_harness {
+/* Tracking operations. */
+enum class tracking_operation { CREATE, INSERT };
+/* Class used to track operations performed on collections */
+class workload_tracking {
+
+    public:
+    workload_tracking(WT_CONNECTION *conn, const std::string &collection_name)
+        : _collection_name(collection_name), _conn(conn), _timestamp(0U)
+    {
+    }
+
+    ~workload_tracking()
+    {
+        if (_session != nullptr) {
+            if (_session->close(_session, NULL) != 0)
+                /* Failing to close session is not blocking. */
+                debug_info(
+                  "Failed to close session, shutting down uncleanly", _trace_level, DEBUG_ERROR);
+            _session = nullptr;
+        }
+    }
+
+    int
+    load(const std::string &table_schema = DEFAULT_TRACKING_TABLE_SCHEMA)
+    {
+        /* Open session. */
+        testutil_check(_conn->open_session(_conn, NULL, NULL, &_session));
+
+        /* Create tracking collection. */
+        testutil_check(_session->create(_session, _collection_name.c_str(), table_schema.c_str()));
+        testutil_check(
+          _session->open_cursor(_session, _collection_name.c_str(), NULL, NULL, &_cursor));
+        debug_info("Tracking collection created", _trace_level, DEBUG_INFO);
+
+        return (0);
+    }
+
+    template <typename K, typename V>
+    int
+    save(const tracking_operation &operation, const std::string &collection_name, const K &key,
+      const V &value)
+    {
+        int error_code;
+
+        _cursor->set_key(_cursor, collection_name.c_str(), key);
```

Comment:
Set_key an set_value both have the potential to return errors, we should capture them.

Reply:
Disregard, they are void.

Reply:
They are void functions